Baikal/PostEffects/ML/buffer.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <tuple>
+
+
+namespace Baikal
+{
+    namespace PostEffects
+    {
+        class Buffer
+        {
+        public:
+            using ValueType = float;
+            using Data = std::shared_ptr<ValueType[]>;
```

Comment:
`std::shared_ptr<T[]>` only works correctly since C++17. See [remarks ](https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr) for point 3), Additionally.

Reply:
Should be solver by using custom deleters e.g. `[](float* data) { delete[] data; }`.